AppCenterCrashes/AppCenterCrashes/MSCrashes.mm

```diff
@@ -262,10 +273,31 @@ - (instancetype)init {
 
 #if TARGET_OS_OSX
     /*
-     * AppKit is preventing applications from crashing on macOS so PLCrashReport cannot catch any crashes.
-     * Setting this flag will let application crash on uncaught exceptions.
+     * Exceptions on the main thread of a Cocoa application do not typically rise to the level of
+     * the uncaught exception handler because the global application object catches all such exceptions.
+     * See: https://developer.apple.com/library/archive/documentation/Cocoa/Conceptual/Exceptions/Concepts/UncaughtExceptions.html
      */
-    [[NSUserDefaults standardUserDefaults] registerDefaults:@{ @"NSApplicationCrashOnExceptions" : @YES }];
+    NSNumber *crashForwarderValue = [[NSBundle mainBundle] objectForInfoDictionaryKey:kMSIsAppCenterCrashForwarderEnabledKey];
+    BOOL crashForwarderEnabled = crashForwarderValue ? [crashForwarderValue boolValue] : YES;
+    if (crashForwarderEnabled) {
+      SEL selector = @selector(reportException:);
+      Method method = class_getInstanceMethod([NSApplication class], selector);
+      IMP implementation = class_getMethodImplementation([MSCrashes class], selector);
+      if (method && implementation) {
+        method_setImplementation(method, implementation);
```

Comment:
Is it possible that the system would be calling `reportException` at the time the swizzling happens? That would cause an undefined behavior I guess.

Reply:
1. It used only for reporting exception from the main thread.
2. **BOTH** implementations will report the crash to the AppCenter. But swizzled one allows get exception information instead of `SIGILL`.

Reply:
And the Crashes service init happens on the main thread so both the wiring the swizzling and the reportException can't happen at the same time.